crates/meroctl/src/cli/context/watch.rs

```diff
@@ -90,25 +91,25 @@ impl WatchCommand {
 
         let mut event_count = 0;
         while let Some(message) = read.next().await {
-            if let Some(max_count) = self.count {
-                if event_count >= max_count {
-                    break;
-                }
-            }
-
             match message {
                 Ok(msg) => {
                     if let WsMessage::Text(text) = msg {
                         let response = serde_json::from_str::<Response>(&text)?;
                         environment.output.write(&response);
 
                         if let Some(cmd) = &self.exec {
-                            let output = Command::new("sh")
-                                .arg("-c")
-                                .arg(cmd)
-                                .output()
-                                .await
-                                .map_err(|e| eyre::eyre!("Failed to execute command: {}", e))?;
+                            if let Some(max_count) = self.count {
+                                if event_count >= max_count {
+                                    break;
+                                }
+                            }
+
+                            let output = if cmd.len() == 1 {
+                                Command::new(&cmd[0]).output().await
+                            } else {
+                                Command::new(&cmd[0]).args(&cmd[1..]).output().await
+                            }
```

Comment:
you shouldn't need distinction between 1 and N, clap should enforce the presence of at least one entry in the vec, but you can test this

Reply:
I've removed the redundant length check